src/univariate/affine.jl

```diff
@@ -0,0 +1,332 @@
+    return @. d.μ + d.σ * extremes
+end
+
+
+## Multivariate
+# when finding the extrema, we need to flip the extremes if the scale is negative
+_flip(signed::Real, tuple::Tuple) = signed > 0 ? tuple : (last(tuple), first(tuple))
+
+function minimum(d::MultivariateAffine)
+    extremes = extrema(d.ρ) |> zip |> collect |> permutedims
+    # The minimum of d.ρ[i] will minimize d[j] if d.σ[i,j] is negative. 
+    # Otherwise the maximum will minimize d[j]
+    extreme_matrix = _flip.(d.σ, extremes)
+    argmins = first.(extreme_matrix)
+    mins = [σ ⋅ argmin for (σ, argmin) in zip(eachrow(d.σ), eachrow(argmins))]
+    return mins
+end
+
+function maximum(d::MultivariateAffine)
+    extremes = extrema(d.ρ) |> zip |> collect |> permutedims
+    extreme_matrix = _flip.(d.σ, extremes)
+    argmaxes = last.(extreme_matrix)
+    maxes = [σ ⋅ argmax for (σ, argmax) in zip(eachrow(d.σ), eachrow(argmaxes))]
+    return maxes
+end
+
+function extrema(d::MultivariateAffine)
+    extremes = extrema(d.ρ) |> zip |> collect |> permutedims
+    extreme_matrix = _flip.(d.σ, extremes)
+    argmins = first.(extreme_matrix)
+    mins = [σ ⋅ argmin for (σ, argmin) in zip(eachrow(d.σ), eachrow(argmins))]
+    argmaxes = last.(extreme_matrix)
+    maxes = [σ ⋅ argmax for (σ, argmax) in zip(eachrow(d.σ), eachrow(argmaxes))]
+    return (mins, maxes)
+end
+
+support(d::AffineDistribution) = affine_support(d.μ, d.σ, support(d.ρ))
+affine_support(μ::Real, σ::Real, support) = μ .+ σ .* support
+
+function affine_support(μ::Real, σ::Real, support::RealInterval) 
+    if σ > 0
+        lower = support.lb
+        upper = support.ub
+    else
+        lower = support.ub
+        upper = support.lb
+    end
+    return RealInterval(μ + σ * lower, μ + σ * upper)
+end
+
+
+#### Conversions
+
+function convert(::Type{AffineDistribution{F,S,Tμ,Tσ,D}}, d::AffineDistribution) where {
+        F<:VariateForm,
+        S<:ValueSupport,
+        D<:Distribution{F, S},
+        Tμ<:ArrayLike,
+        Tσ<:ArrayLike
+    }
+    return AffineDistribution(Tμ(d.μ), Tσ(d.σ), convert(D, d.ρ))
+end
+
+
+function Base.eltype(::Type{<:AffineDistribution{F,S,Tμ,Tσ,D}}) where {
+        F<:VariateForm,
+        S<:ValueSupport,
+        D<:Distribution{F, S},
+        Tμ<:ArrayLike,
+        Tσ<:ArrayLike
+    }
+    T = Core.Compiler.return_type(Base.:*, (Tσ, eltype(D)))
+    return Core.Compiler.return_type(Base.:+, (Tμ, T))
+end
+
+
+#### Parameters
+
+function location(d::AffineDistribution)
+    if hasmethod(location, (typeof(d.ρ),))
+        return d.μ + d.σ * location(d.ρ)
+    else
+        return d.μ
+    end
+end
+function scale(d::AffineDistribution)
+    if hasmethod(scale, (typeof(d.ρ),))
+        return d.σ * scale(d.ρ)
+    else
+        return d.σ
+    end
+end
+shape(d::AffineDistribution) = shape(d.ρ)
+ncategories(d::AffineDistribution) = ncategories(d.ρ)
+params(d::AffineDistribution) = (d.μ, d.σ, d.ρ)
+partype(::AffineDistribution{F,S,Tμ,Tσ,D}) where {F,S,Tμ,Tσ,D} = (Tμ, Tσ)
```

Comment:
The convention is to return a single type (and in particular a type - not a tuple).

Reply:
Not clear to me how this can be dealt with; I switched to using `promote` for now, but what should be done if these parameters have different types (which they often will)?